internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -105,14 +122,14 @@ func SetupWithManager(mgr manager.Manager, opts ...Option) error {
 	}
 
 	controllerName := fmt.Sprintf("controller.bundledeployment.%s", c.provisionerID)
+	l := mgr.GetLogger().WithName(controllerName)
 	controller, err := ctrl.NewControllerManagedBy(mgr).
 		Named(controllerName).
-		For(&rukpakv1alpha1.BundleDeployment{}, builder.WithPredicates(
+		For(&rukpakv1alpha2.BundleDeployment{}, builder.WithPredicates(
 			util.BundleDeploymentProvisionerFilter(c.provisionerID)),
 		).
-		Watches(&source.Kind{Type: &rukpakv1alpha1.Bundle{}}, handler.EnqueueRequestsFromMapFunc(
-			util.MapBundleToBundleDeploymentHandler(context.Background(), mgr.GetClient(), c.provisionerID)),
-		).
+		Watches(source.NewKindWithCache(&corev1.Pod{}, systemNsCache), util.MapOwneeToOwnerProvisionerHandler(context.Background(), mgr.GetClient(), l, c.provisionerID, &rukpakv1alpha2.BundleDeployment{})).
```

Comment:
Not sure if you want to fix this here/now, but we shouldn't be passing in context.Background. We have some options, but they might be worth doing separately (?)

Reply:
I can create a separate issue to handle in a follow up. 

Just curious, why not pass the same context? Isn't it helpful to pass deadlines, cancellations and any request scoping information across API calls?

Reply:
context.Background() never gets canceled, so passing it in here is a no-op.

A map function ideally runs and returns quickly. When using typed generated cache listers, you can traverse from the watch event to whatever you want from the listers without needing a context. But with controller-runtime's Client, you need to pass in a context (and you don't know if you're reading from the cache or using a live client unless you trace the creation of the client you're using).

Let's say instead of context.Background() here, you passed in some cancelable context. Because this call is creating a map function, whatever context you pass in is shared by all invocations of the map function. You can't easily cancel one without canceling all of them (although it's not really clear if we'd want to cancel just one).

Reply:
Thanks for explaining. Have created https://github.com/operator-framework/rukpak/issues/799 for follow up.

Reply:
This will be fixed when we upgrade to controller-runtime v0.16.3.